src/engine/guidance/collapse_scenario_detection.cpp

```diff
@@ -123,7 +123,7 @@ bool isStaggeredIntersection(const RouteStepIterator step_prior_to_intersection,
 
     const auto angle = [](const RouteStep &step) {
         const auto &intersection = step.intersections.front();
-        const auto entry_bearing = intersection.bearings[intersection.in];
+        const auto entry_bearing = util::bearing::reverse(intersection.bearings[intersection.in]);
```

Comment:
Can you elaborate a bit on how you verified that this is necessary?

Off the top of my head and without having checked again:

The bearings for an intersection could look like `[0,90,180,270]` while `in = 0, out = 2`.

This would represent a straight traversal on the north-south axis (note that the bearings are stored from the center of the intersection, so in is already reversed).

The angle between in and out in this representation should be 180 degree, which would be the case without reversing the bearing.
(see `is_right: > 45 < 135`).
With reversing, both bearings would be identical and there would be a turn angle of zero.

I might be totally off here, though.